src/Raven.Server/Utils/Enumerators/PulsedEnumerationState.cs

```diff
@@ -9,13 +9,12 @@ public abstract class PulsedEnumerationState<T>
         internal const int NumberOfEnumeratedDocumentsToCheckIfPulseLimitExceeded = 1024;
 
         protected readonly DocumentsOperationContext Context;
-
-        private readonly Size _pulseLimit;
+        protected Size PulseLimit { get; private set; }
```

Comment:
I don't think those changes are required if we call the `base`

Reply:
Done